pkg/devfile/parser/data/v2/commands_test.go

```diff
@@ -300,3 +301,102 @@ func TestDevfile200_UpdateCommands(t *testing.T) {
 		})
 	}
 }
+
+func TestDeleteCommands(t *testing.T) {
+
+	tests := []struct {
+		name            string
+		commandToDelete string
+		commands        []v1.Command
+		wantCommands    []v1.Command
+		wantErr         bool
+	}{
+		{
+			name:            "Commands that belong to Composite Command",
+			commandToDelete: "command1",
+			commands: []v1.Command{
+				{
+					Id: "command1",
+					CommandUnion: v1.CommandUnion{
+						Exec: &v1.ExecCommand{},
+					},
+				},
+				{
+					Id: "command2",
+					CommandUnion: v1.CommandUnion{
+						Exec: &v1.ExecCommand{},
+					},
+				},
+				{
+					Id: "command3",
+					CommandUnion: v1.CommandUnion{
+						Composite: &v1.CompositeCommand{
+							Commands: []string{"command1", "command2", "command1"},
+						},
+					},
+				},
+			},
+			wantCommands: []v1.Command{
+				{
+					Id: "command2",
+					CommandUnion: v1.CommandUnion{
+						Exec: &v1.ExecCommand{},
+					},
+				},
+				{
+					Id: "command3",
+					CommandUnion: v1.CommandUnion{
+						Composite: &v1.CompositeCommand{
+							Commands: []string{"command1", "command2", "command1"},
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:            "Missing Command",
+			commandToDelete: "command34",
+			commands: []v1.Command{
+				{
+					Id: "command1",
+					CommandUnion: v1.CommandUnion{
+						Exec: &v1.ExecCommand{},
+					},
+				},
+			},
+			wantCommands: []v1.Command{
+				{
+					Id: "command1",
+					CommandUnion: v1.CommandUnion{
+						Exec: &v1.ExecCommand{},
+					},
+				},
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			d := &DevfileV2{
```

Comment:
can we have the devObj defined outside? so we do not need to define commands in each of the test case.
and the test args can be 
```
	tests := []struct {
		name            string
		commandToDelete string
		wantCommands    []v1.Command
		wantErr         bool
	}
```